### PR TITLE
new client security and refactors and playtimes and---

### PR DESCRIPTION
**Pull request by silicons**
cleans up config
adds new role time.tracking system
adds new panic bunker
adds new VPN detection system
unfucks the automated age gate
does a bunch more cleanups
refactors client security checks a little

### CODE REVIEW

File: code/controllers/subsystem/playtime.dm [hunk 1]
```diff
@@ -0,0 +1,305 @@
+/**
+ * playtime tracking system
+ */
+SUBSYSTEM_DEF(playtime)
```
Comment (Kapu1178) [inline code/controllers/subsystem/playtime.dm:4]: this is the most derranged and wasteful way i have ever seen someone impliment a playtime tracker

File: code/modules/client/client.dm [hunk 2]
```diff
@@ -213,3 +213,34 @@
...
+ * are we a guest account?
+ */
+/client/proc/is_guest()
+	return IsGuestKey(key)
+
+/**
+ * are we localhost?
+ */
+/client/proc/is_localhost()
+	return isnull(address) || (address in list("127.0.0.1", "::1"))
```
Comment (Lohikar) [inline code/modules/client/client.dm:240]: BYOND doesn't support IPv6.

Comment (silicons) [reply code/modules/client/client.dm:240]: oh

**Review by TheLordME — APPROVED** [review]

**Review by timothyteakettle — APPROVED** [review]
SQL changes are sane and the ipintel checks look fine

**Pull request merged**